tfhe-benchmark/benches/core_crypto/ks_bench.rs

```diff
@@ -588,157 +591,199 @@ mod cuda {
-                        bench_group.bench_function(&bench_id, |b| {
-                            b.iter(|| {
-                                cuda_keyswitch_lwe_ciphertext_list_into_glwe_ciphertext_64(
-                                    gpu_keys.pksk.as_ref().unwrap(),
-                                    &d_input_lwe_list,
-                                    &mut d_output_glwe,
-                                    &streams,
-                                );
-                                black_box(&mut d_input_lwe_list);
-                            })
-                        });
+                    if !skip_bench {
+                        let gpu_keys = CudaLocalKeys::from_cpu_keys(&cpu_keys, None, &streams);
+
+                        let mut input_ct_list = LweCiphertextList::new(
+                            Scalar::ZERO,
+                            lwe_sk.lwe_dimension().to_lwe_size(),
+                            LweCiphertextCount(glwe_sk.polynomial_size().0),
+                            ciphertext_modulus,
+                        );
+
+                        let plaintext_list = PlaintextList::new(
+                            Scalar::ZERO,
+                            PlaintextCount(input_ct_list.lwe_ciphertext_count().0),
+                        );
+
+                        encrypt_lwe_ciphertext_list(
+                            &lwe_sk,
+                            &mut input_ct_list,
+                            &plaintext_list,
+                            params.lwe_noise_distribution.unwrap(),
+                            &mut encryption_generator,
+                        );
+
+                        let mut d_input_lwe_list = CudaLweCiphertextList::from_lwe_ciphertext_list(
+                            &input_ct_list,
+                            &streams,
+                        );
+
+                        let mut d_output_glwe = CudaGlweCiphertextList::new(
+                            glwe_sk.glwe_dimension(),
+                            glwe_sk.polynomial_size(),
+                            GlweCiphertextCount(1),
+                            ciphertext_modulus,
+                            &streams,
+                        );
+
+                        streams.synchronize();
+
+                        {
+                            bench_group.bench_function(&bench_id, |b| {
+                                b.iter(|| {
+                                    cuda_keyswitch_lwe_ciphertext_list_into_glwe_ciphertext_64(
+                                        gpu_keys.pksk.as_ref().unwrap(),
+                                        &d_input_lwe_list,
+                                        &mut d_output_glwe,
+                                        &streams,
+                                    );
+                                    black_box(&mut d_input_lwe_list);
+                                })
+                            });
+                        }
                     }
```

Comment:
If skip_bench = true maybe we could print a warning message indicating that the bench was skipped due to lack of memory?

Reply:
will the warning show ? I had a println but it only shows when I use `--nocapture` in the cargo command

Reply:
Hmm you're right, it would only show with nocapture. Well that's ok let's leave this as is then.